enterprise/frontend/src/metabase-enterprise/metabot/metabot.unit.spec.tsx

```diff
@@ -0,0 +1,257 @@
+import { combineReducers } from "@reduxjs/toolkit";
+import userEvent from "@testing-library/user-event";
+import fetchMock from "fetch-mock";
+import { P, isMatching } from "ts-pattern";
+
+import { setupEnterprisePlugins } from "__support__/enterprise";
+import { mockSettings } from "__support__/settings";
+import { act, renderWithProviders, screen, waitFor } from "__support__/ui";
+import { useRegisterMetabotContextProvider } from "metabase/metabot";
+import { createMockTokenFeatures } from "metabase-types/api/mocks";
+import { createMockState } from "metabase-types/store/mocks";
+
+import { Metabot } from "./components/Metabot";
+import { MetabotProvider } from "./context";
+import {
+  type MetabotState,
+  metabotInitialState,
+  metabotReducer,
+  setVisible,
+} from "./state";
+
+function setup(
+  options: {
+    ui: React.ReactElement;
+    metabotPluginInitialState?: MetabotState;
+  } | void,
+) {
+  const settings = mockSettings({
+    "token-features": createMockTokenFeatures({
+      metabot_v3: true,
+    }),
+  });
+
+  setupEnterprisePlugins();
+
+  const {
+    ui = <Metabot />,
+    metabotPluginInitialState = { ...metabotInitialState, visible: true },
+  } = options || {};
+
+  return renderWithProviders(<MetabotProvider>{ui}</MetabotProvider>, {
+    storeInitialState: createMockState({
+      settings,
+      plugins: {
+        metabotPlugin: metabotPluginInitialState,
+      },
+    } as any),
+    customReducers: {
+      plugins: combineReducers({
+        metabotPlugin: metabotReducer,
+      }),
+    },
+  });
+}
+
+const chat = () => screen.findByTestId("metabot-chat");
+const input = () => screen.findByTestId("metabot-chat-input");
+const enterChatMessage = async (message: string, send = true) =>
+  userEvent.type(await input(), `${message}${send ? "{Enter}" : ""}`);
+const closeChatButton = () => screen.findByTestId("metabot-close-chat");
+
+// NOTE: for some reason the keyboard shortcuts won't work with tinykeys while testing, using redux for now...
```

Comment:
😬

Reply:
I have some grey hairs from this. I'll bug you or someone tomorrow, maybe I was doing something stupid.